src/MultiIndices/FixedMultiIndexSet.cpp

```diff
@@ -244,6 +245,25 @@ std::vector<unsigned int> FixedMultiIndexSet<Kokkos::HostSpace>::IndexToMulti(un
 }
 
 
+template<typename MemorySpace>
+std::vector<unsigned int> FixedMultiIndexSet<MemorySpace>::NonzeroDiagonalEntries() const
+{
+    assert(false);
+    return std::vector<unsigned int>();
+}
+
+template<>
+std::vector<unsigned int> FixedMultiIndexSet<Kokkos::HostSpace>::NonzeroDiagonalEntries() const
+{
+    std::vector<unsigned int> output;
+    if(!isCompressed) throw std::runtime_error("NonzeroDiagonalEntries only works for compressed multiindex sets");
```

Comment:
I think this function should work whether or not the `FixedMultiIndex` is compressed.  If compressed, can we just revert to a nested loop?  It would be slower, but would work.

Reply:
Yeah, that's fine. I just didn't know what an uncompressed fmset looks like; there's no documentation on it and I wasn't sure where it was used. I figured it was better to write a function that succeeds on most cases then explicitly fails for sure on something that I don't understand than write a function that might or might not fail due to my lack of understanding of the corner case.

Reply:
If I remember correctly, the "compressed" flag just indicates that any 0.0 values in the array of nonzeros have been removed.

Reply:
Should've marked this as resolved; it's been added!